compiler/src/compilers/builtins.rs

```diff
@@ -63,3 +71,145 @@ pub fn compile_builtins_io(compiler: &mut Compiler, idents: Vec<IdentBuiltin>) {
+      },
+      "proc_raise" => IdentType::Function {
+        params: vec![IdentTyped {
+          ident: String::from("sig"),
+          type_ident: IdentType::Primitive(Primitive::I32),
+        }],
+        ret_type: Box::new(IdentType::Primitive(Primitive::None)),
+      },
+      "sched_yield" => IdentType::Function {
+        params: vec![],
+        ret_type: Box::new(IdentType::Primitive(Primitive::I32)),
+      },
+      "args_get" => IdentType::Function {
+        params: vec![
+          IdentTyped {
+            ident: String::from("argvOffset"),
+            type_ident: IdentType::Primitive(Primitive::I32),
+          },
+          IdentTyped {
+            ident: String::from("argvBufferOffset"),
+            type_ident: IdentType::Primitive(Primitive::I32),
+          },
+        ],
+        ret_type: Box::new(IdentType::Primitive(Primitive::None)),
+      },
+      "args_sizes_get" => IdentType::Function {
+        params: vec![
+          IdentTyped {
+            ident: String::from("argcOffset"),
+            type_ident: IdentType::Primitive(Primitive::I32),
+          },
+          IdentTyped {
+            ident: String::from("argvBufferSizeOffset"),
+            type_ident: IdentType::Primitive(Primitive::I32),
+          },
+        ],
+        ret_type: Box::new(IdentType::Primitive(Primitive::I32)),
+      },
+      "environ_get" => IdentType::Function {
+        params: vec![
+          IdentTyped {
+            ident: String::from("environOffset"),
+            type_ident: IdentType::Primitive(Primitive::I32),
+          },
+          IdentTyped {
+            ident: String::from("environBufferOffset"),
+            type_ident: IdentType::Primitive(Primitive::I32),
+          },
+        ],
+        ret_type: Box::new(IdentType::Primitive(Primitive::None)),
+      },
+      "environ_sizes_get" => IdentType::Function {
+        params: vec![
+          IdentTyped {
+            ident: String::from("environcOffset"),
+            type_ident: IdentType::Primitive(Primitive::I32),
+          },
+          IdentTyped {
+            ident: String::from("environBufferSizeOffset"),
+            type_ident: IdentType::Primitive(Primitive::I32),
+          },
+        ],
+        ret_type: Box::new(IdentType::Primitive(Primitive::I32)),
+      },
+      "clock_res_get" => IdentType::Function {
+        params: vec![
+          IdentTyped {
+            ident: String::from("id"),
+            type_ident: IdentType::Primitive(Primitive::I32),
+          },
+          IdentTyped {
+            ident: String::from("resolutionOffset"),
+            type_ident: IdentType::Primitive(Primitive::I32),
+          },
+        ],
+        ret_type: Box::new(IdentType::Primitive(Primitive::I32)),
+      },
+      _ => {
+        compiler.throw(CompilerErrorKind::Unimplemented, 0);
+        IdentType::Error
+      }
+    };
+    setup_builtin(compiler, "wasi_unstable", builtin.ident.as_str(), types);
+  }
+}
```

Comment:
This is quite a massive function, we should probably split it out if that is possible and preferably implement it as a core library instead of builtins considering they are not always present in a wasm application.